File: 1/1.java
import java.util.HashMap;
import java.util.*;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] result = new int[2];
        HashMap<Integer, Integer> hash = new HashMap<Integer, Integer>();
	    for(int i = 0; i < nums.length; i++) {
		int current = nums[i];
		int other = target - current;
		if(hash.containsKey(other)) {
		    result[0] = hash.get(other);
		    result[1] = i;
		    return result;
		    }
		else {
		    hash.put(nums[i],i);
		}
	    }
	    return result;
     }
}

File: 1/1.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hash = {};
        
        for i in range(len(nums)):
            current = nums[i]
            other = target - nums[i]
            if other in hash:
                return [hash[other], i]
            else:
                hash[nums[i]] = i
